app/utils/goRoom.js

```diff
@@ -1,13 +1,16 @@
 import Navigation from '../lib/Navigation';
 import RocketChat from '../lib/rocketchat';
 
-const navigate = ({ item, isMasterDetail, ...props }) => {
+const navigate = async({ item, isMasterDetail, ...props }) => {
 	let navigationMethod = Navigation.navigate;
 
 	if (isMasterDetail) {
 		navigationMethod = Navigation.replace;
 	}
 
+	const { room } = await RocketChat.getRoomInfo(item.rid);
+	item.joinCodeRequired = room.joinCodeRequired;
```

Comment:
Move to https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/views/DirectoryView/index.js#L152.